packages/website/docs/06-Advanced/strict-localization.mdx

```diff
@@ -97,32 +97,41 @@ export function useFormatMessage(): (
 <details>
   <summary><code>react-i18next</code> + <code>LocalizedString</code></summary>
 
-```ts title="my-project/app/src/utils/useTranslation.ts"
-import { StringMap, TOptions } from "i18next";
-import { useTranslation as nativeUseTranslation } from "react-i18next";
-import { LocalizedString } from "@buildo/bento-react-components";
+```ts title="my-project/app/src/i18n/i18n.ts"
+import enMessages from "./locales/en.json";
+
+type Primitive = string | boolean | number | null | undefined;
+type MapLeafNodes<Obj, LeafType> = {
+  [Prop in keyof Obj]: Obj[Prop] extends Primitive
+    ? LeafType
+    : Obj[Prop] extends Record<string | number, any>
+    ? MapLeafNodes<Obj[Prop], LeafType>
+    : never;
+};
```

Comment:
By the way, this MapLeafNode works well for an arbitrarily nested structure, but aren't translation strings flat? Maybe this could be a simpler mapped type?

Reply:
Doesn't `i18next` support nested keys as `react-intl` does? e.g.

```
{ 
   "MyView": {
       "MySection": {
          "key1": "Key 1",
          ...
       }
   }
}
```

Maybe we can simplify the `Primitive` type and check only for strings, but I think the nesting is useful

Reply:
I didn't remember that was allowed. Let's leave it like this then

Reply:
yep, it is:
https://www.i18next.com/misc/json-format -> see the "keyDeep"